presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -368,8 +370,18 @@ private boolean isCandidate(Map<ColumnHandle, NullableValue> bindings)
             }
             LookupSymbolResolver inputs = new LookupSymbolResolver(assignments, bindings);
 
+            Object optimized;
+            try {
+                optimized = evaluator.optimize(inputs);
+            }
+            catch (PrestoException e) {
+                // Skip pruning if evaluation fails in a recoverable way.
+                // Failing here can cause spurious query failures for empty partitions.
+                TryFunction.propagateIfUnhandled(e);
+                return true;
```

Comment:
Maybe we could always `return true` (and not use `TryFunction.propagateIfUnhandled(e);`) here? Pick table layout is going to iterate over partitions which could be later filtered out. So partitions that could cause a failure here, could not participate in actual execution. What do you think?

Reply:
I considered that, but thought it is better to be conservative, as it's easy to mask bugs or other unexpected situations. For example, `FUNCTION_NOT_FOUND` or `GENERIC_INTERNAL_ERROR`. Maybe that's ok, but I'd rather start conservative until we see a counterexample.